### DISCUSSION TITLE
Can I load .node files?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#3195] [Q&A]
I'm trying to use native addons for my application. I have a `.node`, but I get the error: "**No loader is configured for ".node" files: app/native/db.linux-x64-gnu.node**".

I know that my application will only run on a node server (& not something like cloudflare pages), so is there a way to forcibly allow loading of .node files?

**Comment by kiliman** [comment]
See https://github.com/remix-run/remix/discussions/2468

**Reply by vedantroy** [reply]
I believe the suggested fix does not work because `.node` files are not regular assets. When we import a `.png` file, for example, the actual import is a string that represents a URL to the file.

This does not work for `.node` files because `.node` files are more akin to Javascript files. I.e., they must be loaded using `require` (they only work in NodeJS), and the result is a object with a bunch of functions.

I'm currently working around this behavior by doing a dynamic require, and that seems to work fine, but maybe Remix can support `.node` files officially?

**Reply by kiliman** [reply]
Ah, I see. I haven't used `.node` files before. I checked to see if esbuild supported it natively. There's an issue for it https://github.com/evanw/esbuild/issues/1051. He suggests creating a plugin. Unfortunately, Remix doesn't support plugins, so this would have be included in the compiler.

**Reply by sergiodxa** [reply]
If they need to be loaded with require, one option could be to load them in your server code using Express or another Node server, then you can pass the value you got from require in the context object to your loaders and actions
